src/Exclusion/ExpressionLanguageExclusionStrategy.php

```diff
@@ -36,24 +36,39 @@ public function __construct(ExpressionEvaluatorInterface $expressionEvaluator)
      */
     public function shouldSkipProperty(PropertyMetadata $property, Context $navigatorContext): bool
     {
-        if (null === $property->excludeIf) {
+        if (null === $property->excludeIf && null === $property->readOnlyIf) {
             return false;
         }
 
         $variables = [
             'context' => $navigatorContext,
             'property_metadata' => $property,
         ];
+        $readOnlyIf = false;
+        $excludeIf = false;
+
         if ($navigatorContext instanceof SerializationContext) {
             $variables['object'] = $navigatorContext->getObject();
         } else {
             $variables['object'] = null;
+
+            if (null !== $property->readOnlyIf) {
+                if (($property->readOnlyIf instanceof Expression) && ($this->expressionEvaluator instanceof CompilableExpressionEvaluatorInterface)) {
+                    $readOnlyIf = $this->expressionEvaluator->evaluateParsed($property->readOnlyIf, $variables);
+                } else {
+                    $readOnlyIf = $this->expressionEvaluator->evaluate($property->readOnlyIf, $variables);
+                }
+            }
         }
 
-        if (($property->excludeIf instanceof Expression) && ($this->expressionEvaluator instanceof CompilableExpressionEvaluatorInterface)) {
-            return $this->expressionEvaluator->evaluateParsed($property->excludeIf, $variables);
+        if (null !== $property->excludeIf) {
```

Comment:
if `$readOnlyIf` is true. you can skip the branch.

Reply:
good idea, added that